src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Elasticsearch/RequestWrapper.cs

```diff
@@ -115,8 +104,9 @@ private static void TryProcessResponse(IAgent agent, object response, ITransacti
 
                 var uri = GetUriFromResponse(response);
 
-                // TODO: need to figure out how to plumb things so that we can set the uri on the segment after it has already been created.
-                // See comments above about extending the experimental API
+                var data = segment.GetExperimentalApi().SegmentData as IDatastoreSegmentData;
```

Comment:
You could also just call `GetExperimentalApi` once if you save it to a local variable.